File: yyxarch/src/main/java/com/yyxnb/yyxarch/http/interceptor/AddCookiesInterceptor.java
package com.yyxnb.yyxarch.http.interceptor;

import android.util.Log;

import com.yyxnb.yyxarch.http.constant.SPKeys;
import com.yyxnb.yyxarch.utils.PreferencesUtils;

import java.io.IOException;
import java.util.HashSet;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 *         请求头里边添加cookie
 */

public class AddCookiesInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
        HashSet<String> preferences = (HashSet<String>) PreferencesUtils.get(SPKeys.COOKIE, new HashSet<String>());
        if (preferences != null) {
            for (String cookie : preferences) {
                builder.addHeader("Cookie", cookie);
                // This is done so I know which headers are being added; this interceptor is used after the normal logging of OkHttp
                Log.v("RxHttpUtils", "Adding Header Cookie--->: " + cookie);
            }
        }
        return chain.proceed(builder.build());
    }

}

File: yyxarch/src/main/java/com/yyxnb/yyxarch/http/interceptor/AddCookiesInterceptor.kt
package com.yyxnb.yyxarch.http.interceptor

import android.util.Log
import com.tencent.mmkv.MMKV
import com.yyxnb.yyxarch.interfaces.ISPKeys
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*

/**
 * 请求头里边添加cookie
 */

class AddCookiesInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        val preferences = MMKV.defaultMMKV().decodeStringSet(ISPKeys.COOKIE, HashSet<String>()) as? HashSet<String>
        if (preferences != null) {
            for (cookie in preferences) {
                builder.addHeader("Cookie", cookie)
                // This is done so I know which headers are being added; this interceptor is used after the normal logging of OkHttp
                Log.v("RxHttpUtils", "Adding Header Cookie--->: $cookie")
            }
        }
        return chain.proceed(builder.build())
    }

}
